src/utils/session-core.js

```diff
@@ -683,33 +685,52 @@ export const updateUserActivity = async (session) => {
  * @param {object} messageObject - An object containing inputs for the the message
- * @param {string} otherPodUsername - The username of the other user you're sending a message to
  * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
  */
 
-export const sendMessageTTL = async (session, messageObject, otherPodUsername) => {
-  const containerUrl = getContainerUrl(session, 'Inbox', 'cross-fetch', otherPodUsername);
+export const sendMessageTTL = async (session, messageObject) => {
+  const { title, message, recipientUsername } = messageObject;
+  const containerUrl = getContainerUrl(session, 'Inbox', 'cross-fetch', recipientUsername);
+  const inboxUrl = getContainerUrl(session, 'Inbox', 'self-fetch');
+
   const senderUsername = session.info.webId.split('profile')[0].split('/')[2].split('.')[0];
+  const recipientWebId = `https://${recipientUsername}.${
+    SOLID_IDENTITY_PROVIDER.split('/')[2]
+  }/profile/card#me`;
+
+  const senderName = await getUserProfileName(session, session.info.webId);
+  const recipientName = await getUserProfileName(session, recipientWebId);
 
   const date = new Date();
   const dateYYYYMMDD = date.toISOString().split('T')[0].replace(/-/g, '');
   const dateISOTime = date.toISOString().split('T')[1].split('.')[0].replace(/:/g, '');
 
-  const newTtlFile = buildThing(createThing({ name: 'document' }))
-    .addDatetime('https://schema.org/uploadDate', date)
-    .addStringNoLocale(SCHEMA_INRUPT.name, messageObject.file.name)
-    .addStringNoLocale(SCHEMA_INRUPT.identifier, messageObject.type)
-    .addStringNoLocale(SCHEMA_INRUPT.endDate, messageObject.date)
-    .addStringNoLocale(SCHEMA_INRUPT.description, messageObject.description)
+  const newMessageTTL = buildThing(createThing({ name: 'message' }))
+    .addDatetime(RDF_PREDICATES.uploadDate, date)
+    .addStringNoLocale(RDF_PREDICATES.title, title)
+    .addStringNoLocale(RDF_PREDICATES.message, message)
+    .build();
+
+  const senderInfo = buildThing(createThing({ name: 'sender' }))
+    .addStringNoLocale(RDF_PREDICATES.sender, senderName)
+    .addUrl(RDF_PREDICATES.url, session.info.webId)
+    .build();
+
+  const recipientInfo = buildThing(createThing({ name: 'recipient' }))
+    .addStringNoLocale(RDF_PREDICATES.recipient, recipientName)
+    .addUrl(RDF_PREDICATES.url, recipientWebId)
     .build();
 
   let newSolidDataset = createSolidDataset();
-  newSolidDataset = setThing(newSolidDataset, newTtlFile);
+  [newMessageTTL, senderInfo, recipientInfo].forEach((thing) => {
+    newSolidDataset = setThing(newSolidDataset, thing);
+  });
 
-  // Generate document.ttl file for container
-  await saveSolidDatasetInContainer(containerUrl, newSolidDataset, {
-    slugSuggestion: `requestPerms-${senderUsername}-${dateYYYYMMDD}-${dateISOTime}.ttl`,
-    contentType: 'text/turtle',
-    fetch: session.fetch
+  [containerUrl, inboxUrl].forEach(async (inbox) => {
+    await saveSolidDatasetInContainer(inbox, newSolidDataset, {
```

Comment:
this will fail if the container "inbox" does not already exist. The function help text recommends a way to handle this:

```
 * The Container at the given URL should already exist; if it does not, you can initialise it first
 * using [[createContainerAt]], or directly save the SolidDataset at the desired location using
 * [[saveSolidDatasetAt]].
 ```

Reply:
Since I've just refactored sendMessageTTL to throw if Promise.all is not resolved, it solves the inbox does not exist problem by giving the user a failed to send error message on PASS.

However, we do need to consider creating a function to generate a new inbox if inbox does not exist or has been deleted. Could open another issue/PR for this.